src/Checkbox.tsx

```diff
@@ -0,0 +1,62 @@
+import styled from 'styled-components'
+import React, {InputHTMLAttributes, ReactElement, useLayoutEffect, useRef} from 'react'
+import sx, {SxProp} from './sx'
+import {COMMON} from './constants'
+
+export type CheckboxProps = {
```

Comment:
We should add a prop for validation status. Something like:
```
validationStatus?: 'error' | 'success'
```

This would style the input to look like a "success" or "error", and set `aria-invalid="true"` if it's an error.

cc @alliethu - am I correct in thinking that `aria-invalid="true"` should be set when the field fails validation?

Reply:
nice, i've added in the aria roles now. We're not going to be able to easily style this like you suggested though, as it's a native checkbox and we've opting for system styles. Perhaps that part can come in on the next iteration.

Reply:
Do we have an issue created for adding validation styles?

Reply:
Not yet, but i'll create one and link it to https://github.com/github/primer/issues/467

Reply:
Issue for adding validation styles: https://github.com/github/primer/issues/495